include/display/cfb.h

```diff
@@ -0,0 +1,238 @@
+
+enum cfb_display_param {
+	CFB_DISPLAY_HEIGH		= 0,
+	CFB_DISPLAY_WIDTH,
+	CFB_DISPLAY_PPT,
+	CFB_DISPLAY_ROWS,
+	CFB_DISPLAY_COLS,
+};
+
+enum cfb_font_caps {
+	CFB_FONT_MONO_VPACKED		= BIT(0),
+	CFB_FONT_MONO_HPACKED		= BIT(1),
+};
+
+struct cfb_font {
+	const void *data;
+	u8_t width;
+	u8_t height;
+	enum cfb_font_caps caps;
+	u8_t first_char;
+	u8_t last_char;
+};
+
+/**
+ * @brief Macro for creating a font entry.
+ *
+ * @param _name   Name of the font entry.
+ * @param _width  Width of the font in pixels
+ * @param _height Heigth of the font in pixels.
+ * @param _caps   Font capabilities.
+ * @param _data   Raw data of the font.
+ * @param _fc     Character mapped to first font element.
+ * @param _lc     Character mapped to last font element.
+ */
+#define FONT_ENTRY_DEFINE(_name, _width, _height, _caps, _data, _fc, _lc)      \
+	static const struct cfb_font _name				       \
+	__attribute__ ((section(".font_entry."))) __attribute__((used)) =      \
+	{								       \
+		.width = _width,					       \
+		.height = _height,					       \
+		.caps = _caps,						       \
+		.data = _data,						       \
+		.first_char = _fc,					       \
+		.last_char = _lc,					       \
+	}
+
+struct cfb_display_config {
+	/** The width of a display in pixels */
+	u16_t x_res;
+
+	/** The height of a display in pixels */
+	u16_t y_res;
+
+	/** Number of pixels per tile, typically 8 */
+	u8_t ppt;
+};
+
+struct c_framebuffer {
+	/** Pointer to a buffer in RAM */
+	u8_t *buf;
+
+	/** Size of the framebuffer */
+	u32_t size;
+
+	/** Pointer to the font entry array */
+	const struct cfb_font *fonts;
+
+	/** Display capabilities*/
+	enum mdotm_display_caps caps;
+
+	/** Resolution of a framebuffer in pixels in X direction */
+	u8_t x_res;
+
+	/** Resolution of a framebuffer in pixels in Y direction */
+	u8_t y_res;
+
+	/** Number of pixels per tile, typically 8 */
+	u8_t ppt;
+
+	/** Number of available fonts */
+	u8_t numof_fonts;
+
+	/** Current font index */
+	u8_t font_idx;
+
+	/** Optional space between the glyphs */
+	s8_t spacing;
+};
+
+struct mdotm_display_api {
+	/** */
+	struct c_framebuffer fb;
+
+	/** */
+	enum mdotm_display_caps (*get_capabilities)(struct device *dev);
+
+	/** */
+	struct cfb_display_config (*get_config)(struct device *dev);
+
+	/** */
+	int (*display_suspend)(struct device *dev);
+
+	/** */
+	int (*display_resume)(struct device *dev);
+
+	/** */
+	int (*display_set_contrast)(struct device *dev, u8_t contrast);
+
+	/** */
+	int (*write_ram)(struct device *dev,
+			 u8_t *data,
+			 u32_t len,
+			 bool update);
+} __packed;
+
+/**
+ * @brief Print a string into the framebuffer.
+ *
+ * @param dev Pointer to device structure for driver instance
+ * @param str String to print
+ * @param x Position in X direction of the beginning of the string
+ * @param y Position in Y direction of the beginning of the string
+ *
+ * @return 0 on success, negative value otherwise
+ */
+int cfb_print(struct device *dev, char *str, u16_t x, u16_t y);
```

Comment:
This definitely belongs to display handling library, not the driver itself (unless this operation is hardware accelerated).